internal/provider/resource_gitlab_group_access_token.go

```diff
@@ -0,0 +1,237 @@
+		Name:   gitlab.String(d.Get("name").(string)),
+		Scopes: stringSetToStringSlice(d.Get("scopes").(*schema.Set)),
+	}
+	if v, ok := d.GetOk("access_level"); ok {
+		accessLevel := accessLevelNameToValue[v.(string)]
+		options.AccessLevel = &accessLevel
+	}
+
+	log.Printf("[DEBUG] create gitlab GroupAccessToken %s (scopes: %s, access_level: %v) for group ID %s", *options.Name, options.Scopes, options.AccessLevel, group)
+
+	if v, ok := d.GetOk("expires_at"); ok {
+		parsedExpiresAt, err := time.Parse("2006-01-02", v.(string))
+		if err != nil {
+			return diag.Errorf("Invalid expires_at date: %v", err)
+		}
+		parsedExpiresAtISOTime := gitlab.ISOTime(parsedExpiresAt)
+		options.ExpiresAt = &parsedExpiresAtISOTime
+		log.Printf("[DEBUG] create gitlab GroupAccessToken %s with expires_at %s for group ID %s", *options.Name, *options.ExpiresAt, group)
+	}
+
+	groupAccessToken, _, err := client.GroupAccessTokens.CreateGroupAccessToken(group, options, gitlab.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Printf("[DEBUG] created gitlab GroupAccessToken %d - %s for group ID %s", groupAccessToken.ID, *options.Name, group)
+
+	tokenId := strconv.Itoa(groupAccessToken.ID)
+	d.SetId(buildTwoPartID(&group, &tokenId))
+	// NOTE: the token can only be read once after creating it
+	d.Set("token", groupAccessToken.Token)
+
+	return resourceGitlabGroupAccessTokenRead(ctx, d, meta)
+}
+
+func resourceGitlabGroupAccessTokenRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	group, tokenId, err := parseTwoPartID(d.Id())
+	if err != nil {
+		return diag.Errorf("Error parsing ID: %s", d.Id())
+	}
+
+	client := meta.(*gitlab.Client)
+
+	groupAccessTokenId, err := strconv.Atoi(tokenId)
+	if err != nil {
+		return diag.Errorf("%s cannot be converted to int", tokenId)
+	}
+
+	log.Printf("[DEBUG] read gitlab GroupAccessToken %d, group ID %s", groupAccessTokenId, group)
+
+	//there is a slight possibility to not find an existing item, for example
+	// 1. item is #101 (ie, in the 2nd page)
+	// 2. I load first page (ie. I don't find my target item)
+	// 3. A concurrent operation remove item 99 (ie, my target item shift to 1st page)
+	// 4. a concurrent operation add an item
+	// 5: I load 2nd page  (ie. I don't find my target item)
+	// 6. Total pages and total items properties are unchanged (from the perspective of the reader)
+
+	page := 1
+	for page != 0 {
+		groupAccessTokens, response, err := client.GroupAccessTokens.ListGroupAccessTokens(group, &gitlab.ListGroupAccessTokensOptions{Page: page, PerPage: 100}, gitlab.WithContext(ctx))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		for _, groupAccessToken := range groupAccessTokens {
+			if groupAccessToken.ID == groupAccessTokenId {
+
+				d.Set("group", group)
+				d.Set("name", groupAccessToken.Name)
+				if groupAccessToken.ExpiresAt != nil {
+					d.Set("expires_at", groupAccessToken.ExpiresAt.String())
```

Comment:
Found another one
```suggestion
					d.Set("expires_at", groupAccessToken.ExpiresAt.Format("2016-01-02"))
```

Reply:
`ExpiresAt` is of type `gitlab.ISOTime` and not `time.Time` as the `CreatedAt`.

That's why it's working: https://github.com/xanzy/go-gitlab/blob/3d128ad803dd28c9bbcb653da9593c08de1e9bd3/types.go#L402